dt-posts/dt-posts.php

```diff
@@ -1340,7 +1345,551 @@ public static function get_post_activity( string $post_type, int $post_id, array
+                                    ];
+                                }
+                            }
+                        }
+                    }
+
+                    // Package any available values to be updated, accordingly; by field type.
+                    if ( !empty( $values ) ){
+                        if ( $reverted['field_type'] == 'communication_channel' ){
+                            $post_updates[$field_key] = $values;
+                        } else {
+                            $post_updates[$field_key] = [
+                                'values' => $values
+                            ];
+                        }
+                    }
+                    break;
+                case 'date':
+                    $revert_obj = array_values( $reverted['values'] )[0] ?? null;
+                    $post_updates[$field_key] = ( !empty( $revert_obj ) && $revert_obj['keep'] ) ? $revert_obj['value'] : '';
+                    break;
+                case 'number':
+                    $number_update_allowed = true;
+
+                    // Ensure to adhere with any min/max bounds, to avoid exceptions!
+                    $number = !empty( $reverted['values'][0] ) ? $reverted['values'][0] : 0;
+                    if ( isset( $post_type_fields[$field_key], $post_type_fields[$field_key]['min_option'] ) && $post_type_fields[$field_key]['min_option'] > $number ){
+                        $number_update_allowed = false;
+                    }
+                    if ( isset( $post_type_fields[$field_key], $post_type_fields[$field_key]['max_option'] ) && $post_type_fields[$field_key]['max_option'] < $number ){
+                        $number_update_allowed = false;
+                    }
+
+                    // Only update if number format is valid.
+                    if ( $number_update_allowed ){
+                        $post_updates[$field_key] = $number;
+                    }
+                    break;
+                case 'user_select':
+                    $user_select_value = $reverted['values'][0];
+                    if ( $user_select_value != 'user-' ){
+                        $post_updates[$field_key] = $user_select_value;
+                    } else {
+                        $post_updates[$field_key] = '';
+                    }
+                    break;
+                case 'text':
+                case 'boolean':
+                case 'textarea':
+                case 'key_select':
+                    $post_updates[$field_key] = $reverted['values'][0] ?? '';
+                    break;
+            }
+        }
+
+        /**
+         * Final home-straight - Submit updates to revert post back to specified state.
+         * Simply return blank, if no updates are to be made.
+         */
+
+        if ( empty( $post_updates ) ){
+            return [];
+        }
+
+        // Ensure revert activity carried out by Revert Bot, with sufficient permissions.
```

Comment:
@kodinkat 
Please explain this section some more.
What does the current user not have permissions to that the revert bot needs?
Is it the connection fields?
---
If needed, how to make this modular for any post type.
Or, do we instead return an error saying "Some pieces require additional permissions"
Or do we do connections in a second call? A connection might have been deletede?

Reply:
@corsacca 

I think this might be a case of poor variable naming on my part; as this block of logic relates to the actual capabilities setup of the `Revert Bot`.

Reply:
@kodinkat why does the bot need those permissions?

Reply:
To access and update fields.

I guess `peoplegroups` could be removed.